v1/common/amqp.go

```diff
@@ -193,115 +193,109 @@ func (ac *AMQPConnector) ErrChan() chan error {
 }
 
 type amqpConnectionManager struct {
-	url         string
-	tlsConfig   *tls.Config
-	conn        *amqp.Connection
-	newConnChan chan struct{}
-	errChan     chan error
-	mu          *sync.RWMutex
-	resetChan   chan struct{}
-	wg          *sync.WaitGroup
+	url       string
+	tlsConfig *tls.Config
+	conn      *amqp.Connection
+	connChan  chan struct{}
+	errChan   chan error
+	mu        *sync.Mutex
 
 	connectionRetryTimeout time.Duration
 	connectionMaxRetries   int
 }
 
 func newAMQPConnectionManager(url string, tlsConfig *tls.Config) *amqpConnectionManager {
-	return &amqpConnectionManager{
-		url:         url,
-		tlsConfig:   tlsConfig,
-		errChan:     make(chan error),
-		mu:          &sync.RWMutex{},
-		newConnChan: make(chan struct{}),
-		resetChan:   make(chan struct{}),
-		wg:          &sync.WaitGroup{},
-
+	m := &amqpConnectionManager{
+		url:       url,
+		tlsConfig: tlsConfig,
+		errChan:   make(chan error),
+		connChan:  make(chan struct{}),
+		mu:        &sync.Mutex{},
 		connectionRetryTimeout: 5 * time.Second,
 		connectionMaxRetries:   3,
 	}
+	close(m.connChan)
+	return m
 }
 
 func (m *amqpConnectionManager) get() (*amqp.Connection, error) {
-	m.mu.RLock()
-
-	if m.conn == nil {
-		m.mu.RUnlock()
-		return m.makeConnection()
-	}
-
+	m.mu.Lock()
+	defer m.mu.Unlock()
 	select {
-	case <-m.newConnChan:
-		m.mu.RUnlock()
-		return m.makeConnection()
+	case <-m.connChan:
```

Comment:
yes, it can be done, but the good thing about the channel is that with it you get
1) easy synchronization on reset
2) easy synchronization on connection close events (see `notifyClose`)